packages/client/components/DashboardAvatars/DashboardAvatars.tsx

```diff
@@ -64,93 +54,107 @@ const StyledButton = styled(PlainButton)({
 }
 
-type Avatar = DashboardAvatars_team['teamMembers'][0]
+type User = DashboardAvatars_team['teamMembers'][0]['user']
 
 const DashboardAvatars = (props: Props) => {
+  const rowRef = useRef<HTMLDivElement>(null)
   const {team} = props
   const {id: teamId, teamMembers} = team
-  const isDesktop = useBreakpoint(Breakpoint.SIDEBAR_LEFT)
   const atmosphere = useAtmosphere()
   const {viewerId} = atmosphere
-  const maxAvatars = isDesktop ? 10 : 6
-  const overflowCount = teamMembers.length > maxAvatars ? teamMembers.length - maxAvatars + 1 : 0
-  const sortedAvatars = useMemo(() => {
-    const connectedAvatars = [] as Avatar[]
-    const offlineAvatars = [] as Avatar[]
-    teamMembers.forEach((avatar) => {
-      const {id: teamMemberId, user} = avatar
-      const {isConnected} = user
-      const {userId} = fromTeamMemberId(teamMemberId)
+  const sortedUsers = useMemo(() => {
+    const connectedUsers = [] as User[]
+    const offlineUsers = [] as User[]
+    teamMembers.forEach((teamMember) => {
+      const {user} = teamMember
+      const {id: userId, isConnected} = user
       if (userId === viewerId) {
-        connectedAvatars.unshift(avatar)
+        connectedUsers.unshift(user)
       } else if (isConnected) {
-        connectedAvatars.push(avatar)
+        connectedUsers.push(user)
       } else {
-        offlineAvatars.push(avatar)
+        offlineUsers.push(user)
       }
     })
-    const sortedAvatars = connectedAvatars.concat(offlineAvatars)
-    return overflowCount === 0 ? sortedAvatars : sortedAvatars.slice(0, maxAvatars - 1)
+    return connectedUsers.concat(offlineUsers)
   }, [teamMembers])
   const {submitting, onError, onCompleted, submitMutation} = useMutationProps()
 
-  const handleClick = (clickedOverflow: boolean) => {
-    if (!submitting) {
-      submitMutation()
-      ToggleTeamDrawerMutation(
-        atmosphere,
-        {teamId, teamDrawerType: 'manageTeam'},
-        {onError, onCompleted}
-      )
-      commitLocalUpdate(atmosphere, (store) => {
-        const viewer = store.getRoot().getLinkedRecord('viewer')
-        const teamMember = viewer?.getLinkedRecord('teamMember', {teamId})
-        const memberInFocus = teamMembers[clickedOverflow ? maxAvatars - 1 : 0]
-        if (!teamMember || !memberInFocus) return
-        const {id: teamMemberId} = memberInFocus
-        teamMember.setValue(teamMemberId, 'manageTeamMemberId')
-      })
-    }
+  const handleClick = (userId?: string) => {
+    if (submitting) return
+    submitMutation()
+    ToggleTeamDrawerMutation(
+      atmosphere,
+      {teamId, teamDrawerType: 'manageTeam'},
+      {onError, onCompleted}
+    )
+    if (!userId) return
+    commitLocalUpdate(atmosphere, (store) => {
+      const viewer = store.getRoot().getLinkedRecord('viewer')
+      const teamMember = viewer?.getLinkedRecord('teamMember', {teamId})
+      if (!teamMember) return
+      const memberInFocus = teamMembers.find((teamMember) => teamMember.user.id === userId)
```

Comment:
-1 `teamMember` here refers to the `teamMember` in the `teamMembers` array, so we don't need to split up the `teamMember` and `memberInFocus` checks. Can revert the change, so: `if (!teamMember || !memberInFocus) return`